composer/utils/eval_client/eval_client.py

```diff
@@ -7,28 +7,36 @@
 """Abstract class for utilities that access and run code on serverless eval clients."""
 
 import abc
-from typing import Dict
+from typing import Dict, List
 
 __all__ = ['EvalClient']
 
 
 class EvalClient(abc.ABC):
     """Abstract class for implementing eval clients, such as LambdaEvalClient."""
 
-    def invoke(self, payload: Dict[str, str]) -> bool:
-        """Invoke a provided dictionary payload to the client.
+    def invoke(self, payload: List[List[List[Dict[str, str]]]]) -> List[List[List[bool]]]:
+        """Invoke a provided batch of dictionary payload to the client.
 
-        For code generation, the payload is a JSON with the following attributes:
-            {
-                'code': <code to be evaluated>,
-                'input': <test input>,
-                'output': <test output>,
-                'entry_point': <entry point>,
+        For code generation, the payload is a list of list of lists of JSONs. The lists are organized in a nested structure, with the outer list
+        being grouped by the prompt. Then, in the nested list for each prompt, the lists are grouped by the generation beam, or generation. In the
```

Comment:
```suggestion
        being grouped by the prompt. Then, in the nested list for each prompt, the lists are grouped by the generation beam or generation. In the
```
nit: grammar

Reply:
also is this complete? I don't get what generation means?

Reply:
fixed